### PR TITLE
[MD] Feature - Test Connection (Draft PR for initial review)

### PR DESCRIPTION
**Pull request by mpabba3003**
### Description
[MD] Feature - Test Connection (Draft PR for initial review)
 
### Issues Resolved
https://github.com/opensearch-project/OpenSearch-Dashboards/issues/2435

### SCREENSHOTS

https://user-images.githubusercontent.com/109986843/201794484-7d15f94d-4f63-4f0b-a4eb-83b974354667.mp4

 
### Check List
- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
  - [ ] `yarn test:ftr`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: src/plugins/data_source/server/routes/test_connection.ts [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+        await dsValidator.validate();
+
+        return response.ok({
+          body: {
+            success: true,
+          },
+        });
+      } catch (err) {
+        return response.customError({
```
Comment (kristenTian) [inline src/plugins/data_source/server/routes/test_connection.ts:66]: Would this still be needed if we add back in the [router.ts](https://github.com/zhongnansu/OpenSearch-Dashboards/blob/efc244663b00e904a07a3722b32f1d4dac96044d/src/core/server/http/router/router.ts#L303) this? @zhongnansu 

```
      // forward data source error
      if (isDataSourceConfigError(e)) {
        return e;
      }
```

**Comment by kristenTian** [comment]
I promised to check on the client side refactor a bit closer, so here goes :p  https://github.com/opensearch-project/OpenSearch-Dashboards/pull/2885/files

File: src/plugins/data_source/server/routes/test_connection.ts (hunk 1 above)
Comment (zhongnansu) [reply src/plugins/data_source/server/routes/test_connection.ts:66]: Don't think it's a good practice to deal with such error in a commonly used module `src/http`

**Comment by zhongnansu** [comment]
> I promised to check on the client side refactor a bit closer, so here goes :p https://github.com/opensearch-project/OpenSearch-Dashboards/pull/2885/files

Why does it need a refactor? Can you explain here on a single thread?

**Review by zhongnansu — COMMENTED** [review]
@mpabba3003 
Some general question: 
- Is there an option to not build a client and then call "ping" or "clusterInfo" api? It's a simple get request without any payload anyways. OpenSearch API by default supports REST. Can we just use some plain http call instead?
- If you use a client, I feel it's better to close it by the end of the singe API call.
- Do you consider adding re-try logic?

**Pull request closed without merging**